provider/pkg/provider/provider.go

```diff
@@ -205,6 +205,12 @@ func (k *kubeProvider) Call(ctx context.Context, req *pulumirpc.CallRequest) (*p
 	return nil, status.Error(codes.Unimplemented, "Call is not yet implemented")
 }
 
+// GetMapping fetches the mapping for this resource provider, if any. A provider should return an empty
+// response (not an error) if it doesn't have a mapping for the given key.
+func (k *kubeProvider) GetMapping(ctx context.Context, req *pulumirpc.GetMappingRequest) (*GetMappingResponse, error) {
```

Comment:
This looks correct once the pulumi SDK dependency is updated. This currently fails to typecheck because the `GetMapping` method isn't present in the current version of the SDK.